src/components/ObjectInput.tsx

```diff
@@ -0,0 +1,96 @@
+/* @jsx jsx */
+import { jsx } from '@emotion/core';
+import * as React from 'react';
+
+import { Box, Button, Flex, Text } from '@stoplight/ui-kit';
+import fromPairs = require('lodash/fromPairs');
+
+import { fieldName, IFormtronControl } from '..';
+import { DraftValue } from './DraftValue';
+import { ValidityIndicator } from './ValidityIndicator';
+
+export const ObjectInput: React.FunctionComponent<IFormtronControl> = ({
+  id,
+  value = {},
+  schema,
+  onChange,
+  fieldComponents,
+  selection,
+}) => {
+  // Make this thing an array
+  const items = [...Object.entries(value)];
+  const defaultValue = schema.default;
+  const KeyWidget = fieldComponents[fieldName(schema.keys)];
+  const ValWidget = fieldComponents[fieldName(schema.values)];
+
+  const splice = (start: number, deleteCount: number, ...vals: any[]) => {
```

Comment:
See my comment in #17

Reply:
See if you like what I did here. (I decided to do it here, then if we merge this PR first it'll resolve the issue in #17)